### PR TITLE
 Issue #2866 Updated docs for virtualization env setup 

### LINKED ISSUE #2866: Reorganize Documentation getting-started/setting-up-virtualization-environment for Linux
As per the discussion in #2117, it would help new comers if the organization of the doc page could be made smoother by adding separate heads for the OS and under them list the driver followed by steps for each distro.

Link: https://docs.openshift.org/latest/minishift/getting-started/setting-up-virtualization-environment.html

I would be up for contributing, if this seems useful to others.

Thanks

### PR DESCRIPTION
**Pull request by rhnvrm**
Fixes #2866

### CODE REVIEW

File: docs/source/getting-started/setting-up-virtualization-environment.adoc [hunk 1]
```diff
@@ -40,6 +43,8 @@ See the appropriate section for your hypervisor and operating system:
 # chmod +x /usr/local/bin/docker-machine-driver-kvm
 ----
 
+After you have set up the binary, proceed to instructions for your linux distribution.
```
Comment (robin-owen) [inline docs/source/getting-started/setting-up-virtualization-environment.adoc]:
```suggestion
Proceed to the instructions for your Linux distribution after you have set up the binary.
```

Comment (robin-owen) [inline docs/source/getting-started/setting-up-virtualization-environment.adoc]: Linux should be capitalized, and we can avoid comma splicing by rearranging the sentence:

```suggestion
Proceed to the instructions for your Linux distribution after you have set up the binary.
```

File: docs/source/getting-started/setting-up-virtualization-environment.adoc [hunk 2]
```diff
@@ -143,13 +148,16 @@ $ sudo virsh net-autostart default
 
 {next-steps}
 
+[[for-macos]]
+== MacOS
```
Comment (robin-owen) [inline docs/source/getting-started/setting-up-virtualization-environment.adoc]:
```suggestion
== macOS
```

**Comment by LalatenduMohanty** [comment]
@rhnvrm Thanks for the PR. Please update the PR commit message as per https://github.com/minishift/minishift/blob/master/CONTRIBUTING.adoc . The required format for the git commit message is `Issue #<issue_no> message`.

**Comment by rhnvrm** [comment]
@LalatenduMohanty @robin-owen have updated the PR.

File: docs/source/getting-started/setting-up-virtualization-environment.adoc [hunk 3]
```diff
@@ -40,6 +43,8 @@ See the appropriate section for your hypervisor and operating system:
 # chmod +x /usr/local/bin/docker-machine-driver-kvm
 ----
 
+Proceed to the instructions for your Linux distribution after you have set up the binary.
```
Comment (LalatenduMohanty) [inline docs/source/getting-started/setting-up-virtualization-environment.adoc]: s/the binary/the `docker-machine-driver-kvm` binary/

Comment (rhnvrm) [reply docs/source/getting-started/setting-up-virtualization-environment.adoc]: updated

**Comment by rhnvrm** [comment]
@LalatenduMohanty updated

**Comment by LalatenduMohanty** [comment]
ok to run

**Comment by LalatenduMohanty** [comment]
ok to test

File: docs/source/getting-started/setting-up-virtualization-environment.adoc [hunk 4]
```diff
@@ -40,6 +43,8 @@ See the appropriate section for your hypervisor and operating system:
 # chmod +x /usr/local/bin/docker-machine-driver-kvm
 ----
 
+Proceed to the instructions for your Linux distribution after you have set up the docker-machine-driver-kvm binary.
```
Comment (robin-owen) [inline docs/source/getting-started/setting-up-virtualization-environment.adoc]: Nitpick: `docker-machine-driver-kvm` should be styled monospace. :)

Comment (rhnvrm) [reply docs/source/getting-started/setting-up-virtualization-environment.adoc]: @robin-owen thanks for pointing it out, I had missed it :+1:

File: docs/source/getting-started/setting-up-virtualization-environment.adoc [hunk 5]
```diff
@@ -40,6 +43,8 @@ See the appropriate section for your hypervisor and operating system:
 # chmod +x /usr/local/bin/docker-machine-driver-kvm
 ----
 
+Proceed to the instructions for your Linux distribution after you have set up the `docker-machine-driver-kvm` binary.
+
```
Comment (praveenkumar) [inline docs/source/getting-started/setting-up-virtualization-environment.adoc]: :+1:

File: docs/source/getting-started/setting-up-virtualization-environment.adoc [hunk 6]
```diff
@@ -321,26 +317,37 @@ For more information, see xref:../troubleshooting/troubleshooting-driver-plugins
 
 . Verify that you pair the virtual switch with a network card (wired or wireless) that is connected to the network.
 
-. Use the configuration option `hyperv-virtual-switch` or startup flag `--hyperv-virtual-switch` to set the name of the external virtual switch you want to use for {project}.
+. Set the environment variable `HYPERV_VIRTUAL_SWITCH` to the name of the external virtual switch you want to use for {project}.
```
Comment (robin-owen) [inline docs/source/getting-started/setting-up-virtualization-environment.adoc]: I don't recall seeing these changes before -- we need the original wording here. Using the `HYPERV_VIRTUAL_SWITCH` environment variable isn't generally recommended; the related configuration option is more appropriate.

This comment applies down to line 350, where the "Next Steps" heading is lowered.

Comment (rhnvrm) [reply docs/source/getting-started/setting-up-virtualization-environment.adoc]: Hey, thanks for catching that. I guess I accidentally branched from an older local master branch. I will fix this soon.

Comment (robin-owen) [reply docs/source/getting-started/setting-up-virtualization-environment.adoc]: Glad I caught it. :)

Thank you for working through all of the change requests!

Comment (praveenkumar) [reply docs/source/getting-started/setting-up-virtualization-environment.adoc]: @rhnvrm can you fix this and resent so we can merge it.

Comment (rhnvrm) [reply docs/source/getting-started/setting-up-virtualization-environment.adoc]: @robin-owen @praveenkumar, fixed this

**Comment by LalatenduMohanty** [comment]
Thanks @rhnvrm

**Comment by rhnvrm** [comment]
Thank you for reviewing! @robin-owen @LalatenduMohanty @praveenkumar

**Pull request merged**